From d8dbdc13dd0b63f00b9f8676e59e09e4b092789f Mon Sep 17 00:00:00 2001
From: Rick Farrington <rfarrington@marvell.com>
Date: Sat, 4 Jul 2020 19:07:41 -0400
Subject: [PATCH 02/17] Documentation: dt: edac: update sdei-ghes/bed-bert
 settings

commit b9a43edf9565243ca811ef2f22d77caf534307ae from
git@git.assembla.com:cavium/WindRiver.linux.git

Change-Id: I652141a55d7786032bb240c0f02c20dc014a7c41
Signed-off-by: Rick Farrington <rfarrington@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/31488
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../bindings/edac/marvell-sdei-ghes.txt       | 48 +++++++++++++++++--
 1 file changed, 45 insertions(+), 3 deletions(-)

diff --git a/Documentation/devicetree/bindings/edac/marvell-sdei-ghes.txt b/Documentation/devicetree/bindings/edac/marvell-sdei-ghes.txt
index 97fc8b2a5254..633167789817 100644
--- a/Documentation/devicetree/bindings/edac/marvell-sdei-ghes.txt
+++ b/Documentation/devicetree/bindings/edac/marvell-sdei-ghes.txt
@@ -1,6 +1,7 @@
-* Marvell OcteonTX SOC sdei-ghes node
+* Marvell OcteonTX SOC sdei-ghes, bed-bert nodes
 
-The sdei-ghes node is defined to describe resources to the sdei-ghes module.
+The sdei-ghes and bed-bert nodes are defined to describe resources to the
+sdei-ghes module.
 
 Abstract
 --------
@@ -12,7 +13,9 @@ The standard GHES driver requires the presence of ACPI tables, and accompanying
 kernel ACPI support, which is not available to systems utilizing Device Tree.
 
 The OcteonTX2 sdei-ghes module provides the required Hardware Error Source Table
-(HEST) to the kernel, allowing the GHES driver to load.
+(HEST) to the kernel, allowing the GHES driver to load.  This module also
+provides the Boot Error Record Table (BERT) to the kernel, which allows for
+reporting of fatal RAS errors to Linux on a subsequent boot.
 
 Additionally, this module presents GHES devices to the system, which allows
 the firmware (ATF) to report RAS errors.
@@ -22,6 +25,9 @@ The following error sources are supported:
   MDC    - OcteonTX Memory Diagnostic Controller
   MCC    - OcteonTX Memory Common Controller
   LMC    - OcteonTX Local Memory DDR4 Controller
+  BERT   - A virtual device which records fatal errors from any of the above
+           into system preserved memory.
+           NOTE: this is located in the 'bed-bert' node.
 
 Device Tree sdei-ghes binding
 -----------------------------
@@ -67,3 +73,39 @@ Example
 			event-id = <0x40000002>;
 		};
 	}
+
+Device Tree bed-bert binding
+-----------------------------
+
+Required properties:
+- compatible	: Shall be "marvell,bed-bert".
+
+Required properties for bert subnode:
+- reg		: Shall contain three entries, one for each of:
+		  - GHES Error Status Address (ACPI 4.0a, sec 17.3.2.6)
+		  - GHES Error Status Block   (ACPI 4.0a, sec 17.3.2.6.1)
+		  - ring buffer for communication with firmware
+
+Example
+-------
+
+	/* Boot Error Data / BERT support */
+	bed-bert {
+		compatible = "marvell,bed-bert";
+		#address-cells = <2>;
+		#size-cells = <2>;
+		memory-region = <&ghes_bert_reserved>;
+		/*
+		 * Range placeholders; these are set dynamically by
+		 * firmware to match 'reg' in 'ghes_bert_reserved'.
+		 */
+		ranges = <0x0 0x0 0x0 0x0 0x0 0x0>,
+			 <0x0 0x0 0x0 0x0 0x0 0x0>,
+			 <0x0 0x0 0x0 0x0 0x0 0x0>;
+		bert {
+			reg = <0x0 0x00000000 0x0 0x008>,
+			      <0x0 0x00000100 0x0 0x100>,
+			      <0x0 0x00000200 0x0 0x800>;
+		};
+	};
+
-- 
2.17.1

